src/onnx/onnx.cpp

```diff
@@ -2105,7 +2112,9 @@ struct onnx_parser
         return literal{{shape_type, dims}, data.begin(), data.end()};
     }
 
-    static shape parse_type(const onnx::TypeProto& t, const unsigned int batch_size)
+    static shape parse_type(const onnx::TypeProto& t,
+                            const std::vector<std::size_t>& input_dims,
+                            std::size_t default_dim_value)
```

Comment:
Maybe `parse_type` should not be static and then we wont need to pass in these parameters.

Reply:
Fixed.